### ISSUE TITLE
nested exists where does not work without id field

### ISSUE THREAD

**Issue by leetmachine** [prisma/prisma-binding#275]
The exists does not work when using a nested where unless the id field is set on the type.

I have the schema:
```
User {
id: ID! 
email: String!
}

Company {
id: ID!
}

Employee {
user: User!
company: Company!
role: String!
}
```
When I tried to use exists on Employee I received a TypeError: "cannot read property 'length' of undefined' at Prisma.ts:79:31

example exists:
``prisma.exists.Employee({ user: { email: 'test@test.com'}});``

I updated Employee in my Primsa datamodel to include id: ID! and then it was successful (even though I did not change my initial query to use/not use the id.

hopes this helps others!

**Comment by maticzav** [comment]
Hey @leetmachine 👋,

Thank you for reporting this! It would be super helpful if you could provide a small reproduction example - Github repository or CodeSandbox, whichever works best for you. 🙂

**Comment by leosuncin** [comment]
I have the same issue and created a repository with the code necessary to reproduce the issue https://github.com/leosuncin/prisma-binding_exists-bug

More specific the issues is raised here https://github.com/leosuncin/prisma-binding_exists-bug/blob/1dd9e9bfd0937c9c4297834a02f33e1ac0c9b4f5/resolvers/query.js#L2-L9

Moreover, I believe that all this issues prisma/prisma-binding#275 prisma/prisma-binding#252 prisma/prisma-binding#215 are related

**Comment by leosuncin** [comment]
I added functional tests to my repository, also I add `id: ID! @unique` to datamodel but in other [branch](https://github.com/leosuncin/prisma-binding_exists-bug/tree/adding-id) and it works, although it should works in both case because Prisma creates by itself the `id` or that what I tough.

PS: If someone knows a better way to run tests with Prisma, I'm all ears.

**Comment by maticzav** [comment]
@leosuncin I agree, however, to my understanding your datamodel doesn't only shape the fields in your database but also portrays the exposed fields. Considering this, not having an `id` field effectively means hiding it in your data model.

**Comment by leosuncin** [comment]
@maticzav sometimes that's the desire behavior hide a big and not human friendly `id` with a slug or username for instance
